README.md

```diff
@@ -45,14 +45,14 @@ You can use this action along with [create-issue-from-file](https://github.com/p
       automated issue
 
 - name: Update last updated report issue
-  if: ${{ steps.last_issue.outputs.has_found == 'true' }}
+  if: ${{ steps.last-issue.outputs.has-found == 'true' }}
```

Comment:
did you have any working example that uses `steps` and `outputs` with names in kebab case? I was wondering if this notation (`steps.foo-bar`) works

Reply:
> did you have any working example that uses `steps` and `outputs` with names in kebab case?

Yes.

<details>
<summary>I have verified this works in a private repo with this workflow:</summary>


```yaml
name: 'Testing - Manually trigger action'
on:
  workflow_dispatch:
    inputs:
      exit-code:
        type: number
        description: Override the EXIT_CODE
      kebab-bool:
        type: boolean
        description: Test kebab-case conditions
        default: true

jobs:
  lint-links-in-docs:
    name: 'Lint - Documentation Links (with issue reporting)'
    runs-on: ubuntu-22.04
    permissions:
      contents: read
      issues: write
    steps:
      - uses: actions/checkout@v3

      - id: link-reporter
        run: |
          echo '**report content here**' > "${{ runner.temp }}/report.md"

          # Only set when an explicit value is given to workflow input,
          EXIT_CODE=${{ github.event.inputs.exit-code }}
          # Otherwise fallback to 0:
          [[ -z ${EXIT_CODE} ]] && EXIT_CODE=0

          # Disabling this bool will set `EXIT_CODE=2`
          [[ ${{ github.event.inputs.kebab-bool }} == 'false' ]] && EXIT_CODE=2

          echo '::echo::on'
          echo "EXIT_CODE: ${EXIT_CODE}"

          # Test kebab-case output condition:
          echo "::set-output name=exit-code::${EXIT_CODE}"

      # Below manages issue creation/updating when failing links are found:

      - name: 'Check for an existing link-reporter issue'
        id: last-issue
        uses: micalevisk/last-issue-action@v1.2
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          state: open
          labels: ci-lint-links

      - name: 'Create a new issue, or update an existing one'
        if: ${{ steps.link-reporter.outputs.exit-code == 2 }}
        uses: peter-evans/create-issue-from-file@v4
        with:
          title: 'ci(lint): Docs have failing links'
          content-filepath: ${{ runner.temp }}/report.md
          issue-number: ${{ steps.last-issue.outputs.issue_number }}
          labels: ci-lint-links, bug
```

</details>

<details>
<summary>If helpful, here is another test step for shell (`run`) behaviour:</summary>

```yaml
      - name: 'Testing Dispatch Inputs'
        # `inputs` context retains proper casting (true==1, false==0):
        if: ${{ inputs.kebab-bool == true }}
        #if: ${{ inputs.kebab-bool == 1 }}
        # `github.event.inputs` context casts bool to string:
        # Docs: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputs
        #if: ${{ github.event.inputs.kebab-bool == 'true' }}
        # Never valid:
        #if: ${{ github.event.inputs.kebab-bool == true }}
        #if: ${{ inputs.kebab-bool == 'true' }}
        run: |
          echo '::echo::on'
          # Neither cast to shell scripts bools (true==0, false==1), always a string bool:
          X_EVENT=${{ github.event.inputs.kebab-bool }}
          X_INPUT=${{ inputs.kebab-bool }}

          # Only both true or both false lines will echo (assuming no step condition above):
          [[ ${X_EVENT} == 'true'  ]] && echo "is true? event:${X_EVENT}"
          [[ ${X_INPUT} == 'true'  ]] && echo "is true? input:${X_INPUT}"
          [[ ${X_EVENT} == 'false' ]] && echo "is false? event:${X_EVENT}"
          [[ ${X_INPUT} == 'false' ]] && echo "is false? input:${X_INPUT}"

          # Always valid (regardless of bool value, treated as string):
          [[ -n ${X_EVENT}    ]] && echo "n x event:${X_EVENT}"
          [[ -n ${X_INPUT}    ]] && echo "n x input:${X_INPUT}"
          [[ ${X_EVENT} -eq 0 ]] && echo "eq0 event:${X_EVENT}"
          [[ ${X_INPUT} -eq 0 ]] && echo "eq0 input:${X_INPUT}"

          # Never valid (always has a bool string value, even when no default configured):
          [[ -z ${X_EVENT}    ]] && echo 'z x event'
          [[ -z ${X_INPUT}    ]] && echo 'z x input'
          [[ ${X_EVENT} -eq 1 ]] && echo "eq1 event:${X_EVENT}"
          [[ ${X_INPUT} -eq 1 ]] && echo "eq1 input:${X_INPUT}"

          # Avoid negative exit status code from false tests of "Never valid" lines
          echo "Done"
```

</details>

---



> I was wondering if this notation (`steps.foo-bar`) is valid because they are not in javascript

I had a similar concern when I first tried with `kebab-case` (prior to this PR). I'm aware of kebab-case not being suitable convention for JS variable names, but it works here.

I noticed that kebab-case doesn't work well for things like ENV in `run` though, unless you explicitly reference the `${{ env.var-name }}` context.

Beyond that, all outputs are strings or null and comparisons are truthy/falsey like in JS. Numbers 0 and 1 can be coerced to bool for comparions (or rather the other way around), but bool outputs will be strings (of true or false) with no special handling there.

So if you want to avoid comparing bools in the conditions as string bools, you could output as 0 or 1. Docs would need to be quite clear about that though, since similar to when not comparing to a string bool value, the user might accidentally get a false-positive :sweat_smile: 

> (and this expression looks like a js expression)

Based on my testing, I believe you're right and it is a JS expression, but the referenced step output is interpolated to the actual value before that expression is processed.